disco_aws_automation/asiaq_cli.py

```diff
@@ -0,0 +1,160 @@
+"""
+Code for a conventional entry-point based command-line interface.
+"""
+
+import argparse
+import os
+from logging import getLogger
+
+import boto3
+
+from . import DiscoVPC, DiscoAWS, read_config
+from .disco_aws_util import read_pipeline_file, graceful
+from .disco_logging import configure_logging
+
+
+class CliCommand(object):
+    """
+    Abstract(ish) base class for CLI subcommand drivers.  Common behaviors of all subcommands
+    can be implemented here, but of course there's no real guarantee that they'll be honored.
+
+    Required elements in subclasses:
+
+    * a static field called DESCRIPTION, which will be used in arg parsers
+    * a class or static method called "init_args", which takes an ArgumentParser as its
+      input and adds to it whatever arguments and options this command needs.
+    """
+
+    DESCRIPTION = "This command has no description.  Perhaps you should add one?"
+
+    def __init__(self):
+        self.logger = getLogger(type(self).__name__)
+
+    def run(self, args):
+        raise Exception("This is an abstact method. Override it so this command does something useful!")
+
+    @classmethod
+    def init_args(cls, parser):
+        """
+        Set up any arguments and options for this command/subcommand.
+        """
+        pass
+
+
+class SandboxCommand(CliCommand):
```

Comment:
To go into a bit more detail: right now all these commands are "encapsulated" in small executable files with some wild inconsistencies in how they work.  Doing away entirely with encapsulation seems dangerous to me, so I propose having a semi-regular structure that gives some predictability and prevents us from just dumping all the code from all those executables into a single giant package of spaghetti.